docs/hosting/composer.md

```diff
@@ -0,0 +1,240 @@
+
+    composer create-project farmos/project
+
+This will create a `composer.json` file in your directory, copied directly from
+the [farmOS project template](https://packagist.org/packages/farmos/project).
+
+You can use this `composer.json` as the starting point for your own project's
+code repository with [Git](https://git-scm.com/):
+
+```shell
+git init
+git add composer.json
+git commit -m 'Intitial commit of my farmOS project.'
+```
+
+It is important to understand that your project is *not* farmOS, but rather
+farmOS is a *dependency* of your project. You can see this clearly by observing
+that `farmos/farmos` is included in your `composer.json` file's `require`
+section. To customize your project, open `composer.json` in a text editor and
+change the metadata (eg: `name`, `description`, `homepage`, etc).
+
+### Building the codebase
+
+To build you farmOS project's codebase, run the following command:
+
+    composer install --no-dev
+
+This will do two very important things:
+
+1. It will download all of your project's dependencies (including farmOS) and
+   build the file structure. At the end, you will have a fully-functional
+   codebase. Configure your web server's document root to point to the `web`
+   subdirectory, and you will be able to open farmOS in a browser.
+2. It will create a `composer.lock` file that contains auto-generated
+   information about your project's dependencies (and their dependencies) with
+   the specific version that were downloaded. This `composer.lock` file should
+   be committed to source control. It ensures that building the project again
+   will always use the same versions of dependencies, for reproducible builds.
+
+### Ignoring files
+
+At this point there will be many files in your directory that should *not* be
+committed to source control. Best practice is to ignore them with a `.gitignore` file.
+
+Example `.gitignore`:
+
+```
+# Ignore Composer-generated files.
+.editorconfig
+.gitattributes
+vendor
+web/*
+
+# farmOS uses this to store OAuth2 keys.
+keys
+
+# This creates an exception for ./web/modules/custom, which can be used for
+# custom module code that is part of this repository.
+!web/modules
+web/modules/*
+!web/modules/custom
+```
+
+### Community modules
+
+farmOS community modules can be added to your project via Composer.
+
+For example, this will add the
+[farmOS Bee](https://www.drupal.org/project/farm_bee) module:
+
+    composer require drupal/farm_bee
+
+Notice that `composer.json` and `composer.lock` are automatically updated.
+Commit these changes to ensure the module will be downloaded every time your
+codebase is built.
+
+### Custom modules
+
+The example `.gitignore` above includes a special rule to exclude the
+`./web/modules/custom` directory. This allows you to include your own custom
+modules within your project and commit them to source control.
+
+For example, create a directory called `./web/modules/custom/mymodule.info.yml`
```

Comment:
Should we recommend a `mymodule` subdirectory? This would make it easier to add additional custom modules in the future. `./web/modules/custom/mymodule/mymodule.info.yml`

Reply:
Oh good catch @paul121!

Fixed here: https://github.com/farmOS/farmOS/pull/648/commits/dad67fdbd7b86c992a0e2abe9e210d7f7c6a873d